multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -114,28 +110,26 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 		klog.V(2).InfoS("Reset lastUpdateTime", "cluster", memberAnnounce.ClusterID)
 		// Reset lastUpdateTime and connectedLeader for this member.
 		data.lastUpdateTime = time.Now()
-		if len(memberAnnounce.LeaderClusterID) == 0 {
-			data.leaderStatus.connectedLeader = v1.ConditionUnknown
-			data.leaderStatus.message = "Not connected to leader yet"
-			data.leaderStatus.reason = ""
+		data.leaderStatus.connectedLeader = v1.ConditionFalse
+		data.leaderStatus.message = fmt.Sprintf("Local cluster is not the leader of member: %v",
+			memberAnnounce.ClusterID)
+		data.leaderStatus.reason = ReasonNotLeader
+		// Check whether this local cluster is the leader for this member.
+		clusterClaim := &multiclusterv1alpha2.ClusterClaim{}
+		if err := r.Get(context.TODO(), types.NamespacedName{Name: multiclusterv1alpha2.WellKnownClusterClaimID, Namespace: req.Namespace}, clusterClaim); err == nil {
+			if clusterClaim.Value == memberAnnounce.LeaderClusterID {
+				data.leaderStatus.connectedLeader = v1.ConditionTrue
+				data.leaderStatus.message = fmt.Sprintf("Local cluster is the leader of member: %v",
+					memberAnnounce.ClusterID)
+				data.leaderStatus.reason = ReasonConnectedLeader
+			}
 		} else {
-			data.leaderStatus.connectedLeader = v1.ConditionFalse
-			data.leaderStatus.message = fmt.Sprintf("Local cluster is not the leader of member: %v",
-				memberAnnounce.ClusterID)
-			data.leaderStatus.reason = ReasonNotLeader
-			// Check whether this local cluster is the leader for this member.
-			clusterClaim := &multiclusterv1alpha2.ClusterClaim{}
-			if err := r.Get(context.TODO(), types.NamespacedName{Name: multiclusterv1alpha2.WellKnownClusterClaimID, Namespace: req.Namespace}, clusterClaim); err == nil {
-				if clusterClaim.Value == memberAnnounce.LeaderClusterID {
-					data.leaderStatus.connectedLeader = v1.ConditionTrue
-					data.leaderStatus.message = fmt.Sprintf("Local cluster is the leader of member: %v",
-						memberAnnounce.ClusterID)
-					data.leaderStatus.reason = ReasonConnectedLeader
-				}
+			if !apierrors.IsNotFound(err) {
+				return ctrl.Result{}, err
 			}
-			// If err != nil, probably ClusterClaims were deleted during the processing of MemberClusterAnnounce.
-			// Nothing to handle in this case and MemberClusterAnnounce will also be deleted soon.
-			// TODO: Add ClusterClaim webhook to make sure it cannot be deleted while ClusterSet is present.
+			// If it's not found error, ClusterClaims were probably deleted during the processing of MemberClusterAnnounce.
```

Comment:
I feel we should revisit the design of Add/RemoveMember() after ClusterSet changes. Why that indirect, but not start/stop memberAnnounce periodic check right after the memberAnnounce is created/deleted in MemberClusterAnnounceReconciler?

Pasted below my earlier comments too.
> 1. Do we really need leader cluster ID in the MemberAnnounce?
> 2. Do we still need all status conditions after we remove election?
> 3. Add/RemoveMember() need not to be called by ClusterSet changes, after we change to auto-update ClusterSet, but we can start to maintain the timerdata once MemberClusterAnnounce is created?

Reply:
Yeah, from member cluster's perspective, I feel we don't have to provide leader's ClusterID as long as the access token is correct. We probably need to refactor the whole ClusterSet part including the ClusterSet spec vs ClusterSet status for joined member.

Reply:
We can decide leader ClusterID later. But please consider 3). The current code is unnecessarily complex.

Reply:
Sure, I will double check this part.

Reply:
Hi @jianjuns  I have refined this part, the MemberClusterAnnounce is responsible to initialize the data now. Could you help to take another look? thanks.